plugins/inputs/http/http.go

```diff
@@ -95,9 +97,6 @@ func (h *HTTP) gatherURL(
 	if err != nil {
 		return err
 	}
-	if body != nil {
-		defer body.Close()
-	}
```

Comment:
I'm sure I'm missing something, but where is the body closed now with this part removed? Is it done by the http-lib?

Reply:
Since this is no longer an io.ReadCloser, now io.Reader, there is no close method and no need to close it